raster-foundry/raster-foundry
# Add browsing of select CMR collections

## Overview

This PR adds browsing of select CMR collections by adding a CMR repository service. This service can easily be expanded to add additional CMR collections.

This adds browsing of:
- Landsat 1-5 MSS
- Landsat 4-5 TM
- Landsat 7 ETM
- Landsat 8 OLI TIRS
- MODIS/Aqua
- MODIS/Terra

Not all of these collections have thumbnails that can be reasonably displayed on a web mercator map such as we have. For all of CMR currently, only the footprints are shown. As such, some modifications were made to repositories to indicate if they should display thumbnails on the map. Additionally, the scene detail modal takes this into account and will only show the thumbnail on a black background if this is the case.

You cannot currently add these to a project (doing so will throw an unimplemented method error).

The feature is behind a feature-flag.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

### Demo

![image](https://user-images.githubusercontent.com/2442245/35688637-9b47266c-073f-11e8-85c8-718631db4a08.png)

![image](https://user-images.githubusercontent.com/2442245/35688666-ac28aa6e-073f-11e8-9601-e7c797bcbf5c.png)

![image](https://user-images.githubusercontent.com/2442245/35688692-baf026f8-073f-11e8-9bea-b8bacef8ec54.png)

![image](https://user-images.githubusercontent.com/2442245/35688729-d26402c8-073f-11e8-9449-17afaef4ad66.png)

![image](https://user-images.githubusercontent.com/2442245/35688760-e7cf4b54-073f-11e8-8b56-dee40332ddeb.png)

![image](https://user-images.githubusercontent.com/2442245/35688832-158fd734-0740-11e8-9891-425b4beccc14.png)

![image](https://user-images.githubusercontent.com/2442245/35688846-1f44cb90-0740-11e8-9727-60ceffc1be4e.png)

![image](https://user-images.githubusercontent.com/2442245/35688867-2a7bcad6-0740-11e8-8da7-68262964aa13.png)

## Notes

There will need to be additional work done around the various collections and determining if we want to display thumbnails for some collections and not for others.

## Testing Instructions

 * See that after running migrations, you can not select CMR as a repository.
 * Update the feature flag to active (key is 'external-source-browse-cmr')
 * See that the CMR repository is now accessible
 * Check that MODIS/Aqua is the default collection
 * Try out browsing the different repositories (some will need the date filter adjusted).

Closes #2928 (somewhat)
Closes #2888 
Closes #2889 
Closes #2890
